packages/cinterop/src/darwinCommon/kotlin/io/realm/interop/RealmInterop.kt

```diff
@@ -369,6 +374,42 @@ actual object RealmInterop {
         }
     }
 
+    actual fun realm_get_list(obj: NativePointer, key: ColumnKey): NativePointer {
+        return CPointerWrapper(realm_wrapper.realm_get_list(obj.cptr(), key.key))
+    }
+
+    actual fun realm_list_size(list: NativePointer): Long {
+        memScoped {
+            val size = alloc<ULongVar>()
+            checkedBooleanResult(realm_wrapper.realm_list_size(list.cptr(), size.ptr))
+            return size.value.toLong()
+        }
+    }
+
+    actual fun <T> realm_list_get(list: NativePointer, index: Long): T {
+        memScoped {
+            val cvalue = alloc<realm_value_t>()
+            checkedBooleanResult(realm_wrapper.realm_list_get(list.cptr(), index.toULong(), cvalue.ptr))
+            return from_realm_value(cvalue)
+        }
+    }
+
+    actual fun <T> realm_list_add(list: NativePointer, value: T) {
+        memScoped {
+            checkedBooleanResult(
+                realm_wrapper.realm_list_add_by_ref(
+                    list.cptr(),
+                    realm_list_size(list).toULong(),
+                    to_realm_value(value).ptr
+                )
+            )
+        }
+    }
+
+    actual fun realm_list_clear(list: NativePointer) {
+        realm_wrapper.realm_list_clear(list.cptr())
```

Comment:
If the C-API call can fail we should `CheckedBooledResult`!?

Reply:
Yeah, forgot about this one.